src/select/select.js

```diff
@@ -192,37 +201,42 @@ class Select extends React.Component<PropsT, StatelessStateT> {
 
   getSelect() {
     const {
-      overrides: {
-        Root: RootOverride,
-        Input: InputOverride,
-        SearchIcon: SearchIconOverride,
-      } = {},
-    } = this.props;
-    const Root = getOverride(RootOverride) || StyledRoot;
-    const Input = getOverride(InputOverride) || StyledInput;
-    const SearchIcon = getOverride(SearchIconOverride) || StyledSearchIcon;
-    const {placeholder} = this.props;
+      Root: {component: Root, props: rootProps},
+      Input: {component: Input, props: inputProps},
+      SearchIcon: {component: SearchIcon, props: searchIconProps},
+      InputContainer: {component: InputContainer, props: inputContainerProps},
+    } = this.getSubComponents();
+    const {placeholder, disabled} = this.props;
     const {selectedOptions} = this.state;
+    const events = disabled
+      ? {
+          onClickCapture: e => e.stopPropagation(),
+        }
+      : {
+          onKeyDown: e => this.handledHotKeys(e),
+          onClick: () => {
+            this.setState({isDropDownOpen: !this.state.isDropDownOpen});
+          },
+        };
     return (
-      <div
-        tabIndex={this.props.tabIndex}
-        onKeyDown={e => this.handledHotKeys(e)}
-        onClick={() => {
-          this.setState({isDropDownOpen: !this.state.isDropDownOpen});
-        }}
-      >
+      <div tabIndex={this.props.tabIndex} {...events}>
         <InputComponent
           disabled={true}
           placeholder={!selectedOptions.length ? placeholder : ''}
           overrides={{
-            Root: Root,
-            Input: Input,
+            Root: {component: Root, props: rootProps},
```

Comment:
We should probably merge overrides here with those passed by customers in props.